CRM/Core/Payment/BaseIPN.php

```diff
@@ -244,25 +246,28 @@ public function cancelled($objects) {
       }
     }
 
-    $addLineItems = FALSE;
     if (empty($contribution->id)) {
       $addLineItems = TRUE;
```

Comment:
@eileenmcnaughton is this needed now? Also can we simplify the if in new line 260?

Reply:
@seamuslee001 no - not needed now - we could probably simplify that IF - but it's in the part of the code that I think is unreachable & which will be by itself in a deprecated function by the end of it

Reply:
oh ```$addLineItems``` IS still used - although it could go - in general I've about 99% sure that ```$contribution->id``` is ALWAYS set by the time the code gets here